odl/deform/optimal_information_transport.py

```diff
@@ -0,0 +1,205 @@
+                                                      bounds_check=False)))
+        DPhiJacobian = np.exp(eps * grad.adjoint(v)) * gradS.domain.element(
+            DPhiJacobian.interpolation(new_points, bounds_check=False))
+
+        if callback is not None:
+            callback(I)
+
+
+# Code for the example starts here
+
+
+def SNR(signal, noise, impl='general'):
+    """Compute the signal-to-noise ratio.
+    This compute::
+        impl='general'
+            SNR = s_power / n_power
+        impl='dB'
+            SNR = 10 * log10 (
+                s_power / n_power)
+    Parameters
+    ----------
+    signal : projection
+    noise : white noise
+    impl : implementation method
+    """
+    if np.abs(np.asarray(noise)).sum() != 0:
+        ave1 = np.sum(signal)/signal.size
+        ave2 = np.sum(noise)/noise.size
+        s_power = np.sqrt(np.sum((signal - ave1) * (signal - ave1)))
+        n_power = np.sqrt(np.sum((noise - ave2) * (noise - ave2)))
+        if impl == 'general':
+            snr = s_power/n_power
+        else:
+            snr = 10.0 * np.log10(s_power/n_power)
+
+        return snr
+
+    else:
+        return float('inf')
+
+
+if __name__ == '__main__':
+    import odl
+
+    I0name = '../../ddmatch/Example3 letters/c_highres.png'
+    I1name = '../../ddmatch/Example3 letters/i_highres.png'
+
+    I0 = np.rot90(plt.imread(I0name).astype('float'), -1)[::2, ::2]
+    I1 = np.rot90(plt.imread(I1name).astype('float'), -1)[::2, ::2]
+
+    # Discrete reconstruction space: discretized functions on the rectangle
+    space = odl.uniform_discr(
+        min_corner=[-16, -16], max_corner=[16, 16], nsamples=[128, 128],
+        dtype='float32', interp='linear')
+
+    # Make a parallel beam geometry with flat detector
+    # Angles: uniformly spaced, n = 360, min = 0, max = 2 * pi
+    # Create the uniformly distributed directions
+    angle_partition = odl.uniform_partition(0, np.pi, 6)
+
+    # Create 2-D projection domain
+    # The length should be 1.5 times of that of the reconstruction space
+    detector_partition = odl.uniform_partition(-24, 24, 192)
+
+    # Create 2-D parallel projection geometry
+    geometry = odl.tomo.Parallel2dGeometry(angle_partition,
+                                           detector_partition)
+
+    # ray transform aka forward projection. We use ASTRA CUDA backend.
+    ray_trafo = odl.tomo.RayTransform(space, geometry, impl='astra_cuda')
+
+    # Create the ground truth as the given image
+    ground_truth = space.element(I0)
+
+    # Create projection data by calling the ray transform on the phantom
+    proj_data = ray_trafo(ground_truth)
+
+    # Add white Gaussion noise onto the noiseless data
+    noise = odl.util.white_noise(ray_trafo.range) * 3
+
+    # Create the noisy projection data
+    noise_proj_data = proj_data + noise
+
+    # Compute the signal-to-noise ratio in dB
+    snr = SNR(proj_data, noise, impl='dB')
+
+    # Output the signal-to-noise ratio
```

Comment:
This is overdocumenting, the code is quite easy to read without these comments. In general comments should tell the reader **why** something is done, especially when it is not obvious. **What** is done should in general be obvious from the code (if it is not, then documenting is needed).

I would write

```python
# Create projection data
proj_data = ray_trafo(ground_truth)
noise_proj_data = proj_data + odl.util.white_noise(ray_trafo.range) * 3
print('snr = {!r}'.format(SNR(proj_data, noise, impl='dB')))
```